File: 01/01.py
import math


def iterate(m):
    return int(math.floor(m / 3)) - 2


def fuel_for_mass(m):
    total = 0
    while m > 0:
        m = iterate(m)
        if m > 0:
            total += m
    return total


total_fuel = 0
partial_fuel = 0
with open('masses.txt') as masses:
    for l in masses.readlines():
        total_fuel += fuel_for_mass(int(l))
        partial_fuel += iterate(int(l))

print 'part 1:', partial_fuel
print 'part 2:', total_fuel

File: 01/01.clj
(ns dec01
  (:use [clojure.string :only [split-lines]]))

(defn parse-file [file-name]
    (map read-string (split-lines (slurp file-name))))

(defn fuel-for-mass [mass]
    (- (int (/ mass 3)) 2))

(defn total-fuel-for-mass [mass]
    (reduce +
        (take-while pos? (rest (iterate fuel-for-mass mass)))))

(let [masses (parse-file "masses.txt")]        
    (println "Part 1:" (reduce + (map fuel-for-mass masses)))
    (println "Part 2:" (reduce + (map total-fuel-for-mass masses))))